transports/p2p/addrmgr/addrmanager.go

```diff
@@ -211,6 +227,48 @@ func (a *AddrManager) updateAddress(netAddr, srcAddr *wire.NetAddress) {
 	a.Log.Trace().Msgf("Added new address %s for a total of %d addresses", addr, a.nTried+a.nNew)
 }
 
+// BanAddress creates a ban for defaultBanTime for a peer that we cannot connect to.
+func (a *AddrManager) BanAddress(addr string) {
+	banTimeUntil := time.Now().Add(banTime)
+	a.addrBanned[addr] = banTimeUntil
+
+	a.removeAddrFromTried(addr)
+	a.removeAddrFromNew(addr)
+
+	a.Log.Warn().Msgf("Address %s banned until %s!", addr, banTimeUntil.Format(time.RFC1123))
+}
+
+func (a *AddrManager) removeAddrFromTried(addr string) {
+	for _, bucket := range a.addrTried {
```

Comment:
#### only comment for the future (resolve when you read it)

This is soooooo hard to read because of used structure, that I would consider those addrTried and addrNew things that could be somewhere at the top of the list of candidates to refactor.

Reply:
100% agree. The difference in structures between addrNew and addrTried and also the address buckets are driving me craazy, they introduce soo much complexity without any significant performance/usefulness benefits.